### ISSUE TITLE
ChainCRF , NSlackSSVM - Tuple index out of range

### ISSUE THREAD

**Issue by samieti02** [pystruct/pystruct#232]
Hello my dear people,

I have some troubles runninng the code 

model = ChainCRF()
ssvm = OneSlackSSVM(model=model, C=.1)
ssvm.fit(X_train, y_train)

where X_train is an array with the shape (70000 , 1)
 and Y_train is another array with shape (70000, )

I always get the error , tuple index out of range
And I tried different variants of the shape, I also transformed array in tuple, but it didnt help.
Can you explain us how it work with the shape of arrays ?

Thanks Sir

**Comment by jlmeunier** [comment]
Hi,
Sorry I never used this model.
But maybe have a look at pystruct/tests/test_models/test_chain_crf.py ?
JL

**Comment by samieti02** [comment]
Hello Sir and thanks for the reply,
I tried with differents shape, but still dont work. In the reference, there is no informations about the type and shape of the dataset.
My data are pixel of 3d image (x,y,z) which I have flattened into a row, this is how we get (70000,1). This input work for example with svm sklearn. 
With pystruct, there is probably another form of input. Can you please help us? And also give us an answer bout the code above?
Thanks

**Comment by subrahmanyap** [comment]
Check if your labels are in the correct range. If there are 4 labels, the labels should be having values 0,1,2,3. For instance, if your labels are 1,2,3,4, it might throw error. Make sure your labels are in correct range. 

Also, check  X[0].shape. It should of the shape (no_of_nodes, no_of_features).
